app/components/primer/layout.rb

```diff
@@ -0,0 +1,347 @@
+    # @param sidebar_width [Symbol] <%= one_of(Primer::Layout::SIDEBAR_WIDTH_OPTIONS) %>
+    # @param sidebar_placement [Symbol] <%= one_of(Primer::Layout::SIDEBAR_PLACEMENT_OPTIONS) %>
+    # @param sidebar_flow_row_placement [Symbol] Sidebar placement when `Layout` is flowing as row. <%= one_of(Primer::Layout::SIDEBAR_FLOW_ROW_PLACEMENT_OPTIONS) %>
+    # @param main_width [Symbol] <%= one_of(Primer::Layout::MAIN_WIDTH_OPTIONS) %>
+    # @param divider [Boolean] Wether or not to add a divider between `main` and `sidebar`.
+    # @param divider_flow_row_variant [Symbol] Variants for the divider when `Layout` is flowing as row. <%= one_of(Primer::Layout::DIVIDER_FLOW_ROW_VARIANT_OPTIONS) %>
+    # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+    def initialize(
+      density: DENSITY_DEFAULT,
+      container: CONTAINER_DEFAULT,
+      gutter: GUTTER_DEFAULT,
+      flow_row_until: FLOW_ROW_UNTIL_DEFAULT,
+      sidebar_width: SIDEBAR_WIDTH_DEFAULT,
+      sidebar_placement: SIDEBAR_PLACEMENT_DEFAULT,
+      sidebar_flow_row_placement: SIDEBAR_FLOW_ROW_PLACEMENT_DEFAULT,
+      main_width: MAIN_WIDTH_DEFAULT,
+      divider: false,
+      divider_flow_row_variant: DIVIDER_FLOW_ROW_VARIANT_DEFAULT,
+      **system_arguments
+    )
+      @container = container
+      @divider = divider
+      @sidebar_placement = fetch_or_fallback(SIDEBAR_PLACEMENT_OPTIONS, sidebar_placement, SIDEBAR_PLACEMENT_DEFAULT)
+      @sidebar_flow_row_placement = fetch_or_fallback(SIDEBAR_FLOW_ROW_PLACEMENT_OPTIONS, sidebar_flow_row_placement, SIDEBAR_FLOW_ROW_PLACEMENT_DEFAULT)
+      @main_width = fetch_or_fallback(MAIN_WIDTH_OPTIONS, main_width, MAIN_WIDTH_DEFAULT)
+
+      @divider_classes = class_names(
+        "Layout-divider",
+        DIVIDER_FLOW_ROW_VARIANT_MAPPINGS[fetch_or_fallback(DIVIDER_FLOW_ROW_VARIANT_OPTIONS, divider_flow_row_variant, DIVIDER_FLOW_ROW_VARIANT_DEFAULT)]
+      )
+
+      @system_arguments = system_arguments
+      @system_arguments[:tag] = :div
+      @system_arguments[:m] = DENSITY_MAPPINGS[fetch_or_fallback(DENSITY_OPTIONS, density, DENSITY_DEFAULT)]
+      @system_arguments[:classes] = class_names(
+        "Layout",
+        "Layout--sidebarPosition-#{@sidebar_placement}",
+        "Layout--sidebarPosition-flowRow-#{@sidebar_flow_row_placement}",
+        system_arguments[:classes],
+        SIDEBAR_WIDTH_MAPPINGS[fetch_or_fallback(SIDEBAR_WIDTH_OPTIONS, sidebar_width, SIDEBAR_WIDTH_DEFAULT)],
+        GUTTER_MAPPINGS[fetch_or_fallback(GUTTER_OPTIONS, gutter, GUTTER_DEFAULT)],
+        FLOW_ROW_UNTIL_MAPPINGS[fetch_or_fallback(FLOW_ROW_UNTIL_OPTIONS, flow_row_until, FLOW_ROW_UNTIL_DEFAULT)],
+        "Layout--divided" => divider
+      )
+    end
+
+    def render?
+      main.present? && sidebar.present?
+    end
+
+    private
+
+    def wrapper
+      if @container == :full
```

Comment:
Isn't this more akin to being `none`, than `full`?

Reply:
I think `full` makes sense given the doc: "If `full`, the Layout component is applied edge-to-edge"